packages/storybook/stories/MessageThread/snippets/MessageWithCustomAttachment.snippet.tsx

```diff
@@ -0,0 +1,103 @@
+import {
+  AttachmentMenuAction,
+  AttachmentMetadata,
+  ChatMessage,
+  FluentThemeProvider,
+  MessageStatus,
+  MessageThread,
+  defaultAttachmentMenuAction
+} from '@azure/communication-react';
+import { initializeFileTypeIcons } from '@fluentui/react-file-type-icons';
+import { ShareIos24Regular, WindowNew24Regular } from '@fluentui/react-icons';
+import React from 'react';
+
+initializeFileTypeIcons();
+
+export const MessageWithCustomAttachment: () => JSX.Element = () => {
+  return (
+    <FluentThemeProvider>
+      <MessageThread
+        userId={'1'}
+        showMessageDate={true}
+        attachmentOptions={{
+          downloadOptions: {
+            actionsForAttachment: (attachment: AttachmentMetadata, message?: ChatMessage): AttachmentMenuAction[] => {
+              // allows the menu actions to be dynamically generated based on the attachment content
+              // (i.e. name, extension, etc.), or the message content (i.e. senderID, etc.)
+              // in this example, we are constructing a dynamic menu based on the attachment extension
+              if (attachment.extension === 'pdf') {
+                return [
+                  {
+                    name: 'Share',
+                    icon: <ShareIos24Regular />,
+                    onClick: () => {
+                      return new Promise((resolve) => {
+                        window.open(attachment.url, '_blank');
+                        resolve();
+                      });
+                    }
+                  }
+                ];
+              } else {
+                return [
+                  defaultAttachmentMenuAction,
+                  {
+                    name: 'Open',
+                    icon: <WindowNew24Regular />,
+                    onClick: () => {
+                      return new Promise((resolve) => {
+                        window.alert('open button clicked for messageID: ' + message?.messageId);
+                        resolve();
+                      });
+                    }
+                  }
+                ];
+              }
+            }
+          }
+        }}
+        messages={[
+          {
+            messageType: 'chat',
+            senderId: 'user1',
+            senderDisplayName: 'Kat Larsson',
+            messageId: Math.random().toString(),
+            content: 'This is the report:',
+            createdOn: new Date('2019-04-13T00:00:00.000+08:10'),
+            mine: false,
+            attached: false,
+            status: 'seen' as MessageStatus,
+            contentType: 'text',
+            attachments: [
+              {
+                id: 'SomeUniqueId1',
+                name: 'Report.docx',
+                extension: 'docx',
+                url: 'https://www.example.com/files/report.docx'
```

Comment:
Can we use anyvalid link here? (It returns 404 right now)

Reply:
no it's supposed to be a mock URL, we shouldn't host any files for user to download imo